Improve git's goods

bogdanvlviv opened the pull request:
#### Improve /.gitignore

#### Remove actionpack/test/tmp/.gitignore


#### Create /activerecord/.gitignore. Clean up /.gitignore
  * Move /activerecord/sqlnet.log to /activerecord/.gitignore
  * Remove /activerecord/examples/.gitignore
  * Move /activerecord/test/.gitignore to /activerecord/.gitignore
  * Move /activerecord/test/fixtures/.gitignore to /activerecord/.gitignore


#### Remove /activemodel/test/fixtures/fixture_database.sqlite3 from /.gitignore


#### Create /activesupport/.gitignore. Clean up /.gitignore
  * Move /activesupport/test/fixtures/isolation_test to /activesupport/.gitignore


#### Improve /railties/.gitignore. Clean up /.gitignore
  * Move /railties/test/500.html, /railties/test/fixtures/tmp,
    /railties/test/initializer/root/log and /railties/tmp to /railties/.gitignore
  * Remove redundant /railties/doc from /.gitignore because it is covered by /*/doc


#### Improve /actionview/.gitignore.
  * Move /actionview/log to actionview/.gitignore
  * Add package-lock.json to/.gitignore
  * Move actionview/test/ujs/.gitignore to actionview/.gitignore
  * Move actionview/test/fixtures/public/.gitignore to actionview/.gitignore


#### Express the folders in /actioncable/.gitignore


#### Clean up /activestorage/.gitignore
  * Remove .byebug_history and node_modules because it is ignored by /.gitignore
  * Move /activestorage/test/service/configurations.yml to /activestorage/.gitignore


#### Remove actionview/test/tmp/.keep

  Related to #11743, #30392 

  ```
  rm actionview/test/tmp/ -fr
  cd actionview/
  bundle exec jruby -Itest test/template/digestor_test.rb
  # All tests in test/template/digestor_test.rb pass on JRuby
  ```


I don't think that it is a good idea to have a lot of `.gitignore` files on a project like `activerecord/examples/.gitignore` because it is hard to manage it.
I see as a good pattern (in this project) to have `/.gitignore` for common things and `/*/.gitignore` for things that belong only this module.

bogdanvlviv committed:
Improve /actionview/.gitignore.

  * Move /actionview/log to actionview/.gitignore
  * Add package-lock.json to/.gitignore
  * Move actionview/test/ujs/.gitignore to actionview/.gitignore
  * Move actionview/test/fixture/s/public/.gitignore to actionview/.gitignore

bogdanvlviv committed:
Improve /railties/.gitignore. Clean up /.gitignore

  * Move /railties/test/500.html, /railties/test/fixtures/tmp,
    /railties/test/initializer/root/log and /railties/tmp to /railties/.gitignore
  * Remove redundant /railties/doc from /.gitignore because it is covered by /*/doc

bogdanvlviv committed:
Create /activesupport/.gitignore. Clean up /.gitignore

  * Move /activesupport/test/fixtures/isolation_test to /activesupport/.gitignore

bogdanvlviv committed:
Remove /activemodel/test/fixtures/fixture_database.sqlite3 from /.gitignore

bogdanvlviv committed:
Create /activerecord/.gitignore. Clean up /.gitignore

  * Move /activerecord/sqlnet.log to /activerecord/.gitignore
  * Remove /activerecord/examples/.gitignore
  * Move /activerecord/test/.gitignore to /activerecord/.gitignore
  * Move /activerecord/test/fixtures/.gitignore to /activerecord/.gitignore

bogdanvlviv committed:
Remove actionpack/test/tmp/.gitignore

bogdanvlviv committed:
Improve /.gitignore

lugray commented on .gitignore:
```diff
@@ -1,24 +1,14 @@
-# Don't put *.swp, *.bak, etc here; those belong in a global ~/.gitignore.
+# Don't put *.swp, *.bak, etc here; those belong in a global ~/.gitignore_global.
```
The default global ignore is `$XDG_CONFIG_HOME/git/ignore`. If `$XDG_CONFIG_HOME` is either not set or empty, `$HOME/.config/git/ignore` is used instead. ([source](https://git-scm.com/docs/gitignore/)) This should either be updated to reflect that, or left as is, but not changed to some arbitrary value.

bogdanvlviv replied:
The line below provides a link.
I just want to be consistent with content that the link provides https://help.github.com/articles/ignoring-files.

matthewd replied:
That's the example name it uses, but the heading is "Create a global .gitignore"

bogdanvlviv replied:
`~/.gitignore` => `.gitignore`

matthewd commented:
> I don't think that it is a good idea to have a lot of .gitignore files on a project like activerecord/examples/.gitignore because it is hard to manage it.

For such locally-relevant ignores (in this case, a file that's generated by another file in that directory), it feels much neater to me to keep it right where it is.

And that file hasn't been changed since it was added 8 years ago, so I'm unconvinced we're suffering greatly from it being "hard to manage". :confused:

bogdanvlviv committed:
Express the folders in /actioncable/.gitignore

bogdanvlviv committed:
Clean up /activestorage/.gitignore

  * Remove .byebug_history and node_modules because it is ignored by /.gitignore
  * Move /activestorage/test/service/configurations.yml to /activestorage/.gitignore

bogdanvlviv commented:
@matthewd Thanks for the review.

I rebased the branch with master and cleaned up `/activestorage/.gitignore`.  

`/activerecord/examples/.gitignore` was needed then became redundant. From this example, I see that it hard to track changes. So I believe that my approach will prevent keeping redundant files and redundant lines in .gitignore files.

Would love to hear more thoughts, advice and made this in the best way for all of us.

bogdanvlviv committed:
Remove actionview/test/tmp/.keep

  Related to #11743, #30392

  ```
  rm actionview/test/tmp/ -fr
  cd actionview/
  bundle exec jruby -Itest test/template/digestor_test.rb
  # All tests in test/template/digestor_test.rb pass on JRuby
  ```
